### DISCUSSION TITLE
Couldn't able to extract value from JSON if the key starts with special character '#'

### DISCUSSION THREAD

**Discussion by VelSankar-DT** [apache/camel-karavan#1103] [Q&A] [answered]
`{
  "name": "raknas",
  "#text": "employee"
}`

I couldn't able to extract "#text" property value using with ${jq(.#text)}

Is there any alternative solution for this?

**Comment by davsclaus** [comment]
that is a bad style to use # in key names. Is there a specific case why you do that

**Reply by VelSankar-DT** [reply]
`<Persons>
<Person name="raknas" age="24">employee</Person>
<Person name="sankar" age="24">admin</Person>
</Persons>`

When I use the 'X Identity Action' for XML to JSON transformation for the above XML format. Those inner texts are shown in the '#text' key. So, I couldn't able to extract that value using ${jq(.#text)}

**Reply by davsclaus** [reply]
You can try with single quotes, jq(.'#text')} to see if that helps

**Reply by davsclaus** [reply]
Okay you need to use double quotes
https://jqlang.github.io/jq/manual/#basic-filters

."#text"

**Comment by davsclaus** [comment]
Also this is not karavan specific but general JQ and Camel

**Comment by davsclaus** [comment]
There is likely a little bug in Camel 4.3 that removes the double quotes
https://issues.apache.org/jira/browse/CAMEL-20392

**Reply by davsclaus** [reply, marked as answer]
Okay with this fix, then it works with double quotes. So you would need Camel 4.4 when its released.

**Comment by davsclaus** [comment]
And btw with camel-jbang you can experiment with message transformation from a CLI / terminal

I put your sample data into file `payload.json` and the mapping code into `template.jq`. And then you can call

```
camel transform message --body=file:payload.json --language=simple --template=file:template.jq
```

You can run this in `--watch` mode then it keeps reloading on-save file, so you can quickly try out and see output.

There are also videos on youtube about this.
